src/test/java/io/github/thebusybiscuit/slimefun4/tests/items/TestCategories.java

```diff
@@ -0,0 +1,187 @@
+package io.github.thebusybiscuit.slimefun4.tests.items;
+
+import java.time.LocalDate;
+import java.time.Month;
+
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import be.seeseemelk.mockbukkit.ServerMock;
+import io.github.thebusybiscuit.cscorelib2.item.CustomItem;
+import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
+import io.github.thebusybiscuit.slimefun4.core.categories.FlexCategory;
+import io.github.thebusybiscuit.slimefun4.core.categories.LockedCategory;
+import io.github.thebusybiscuit.slimefun4.core.categories.SeasonalCategory;
+import io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuideLayout;
+import io.github.thebusybiscuit.slimefun4.mocks.SlimefunMocks;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+public class TestCategories {
+
+    private static ServerMock server;
+    private static SlimefunPlugin plugin;
+
+    @BeforeAll
+    public static void load() {
+        server = MockBukkit.mock();
+        plugin = MockBukkit.load(SlimefunPlugin.class);
+    }
+
+    @AfterAll
+    public static void unload() {
+        MockBukkit.unmock();
+    }
+
+    @Test
+    public void testCategoryGetters() {
+        Category category = new Category(new NamespacedKey(plugin, "getter_test"), new CustomItem(Material.DIAMOND_AXE, "&6Testing"));
+
+        Assertions.assertEquals(3, category.getTier());
+        Assertions.assertEquals(new NamespacedKey(SlimefunPlugin.instance, "getter_test"), category.getKey());
+        Assertions.assertEquals("Testing", category.getUnlocalizedName());
+        Assertions.assertEquals(0, category.getItems().size());
+    }
+
+    @Test
+    public void testAddItem() {
+        Category category = new Category(new NamespacedKey(plugin, "items_test"), new CustomItem(Material.DIAMOND_AXE, "&6Testing"));
+        SlimefunItem item = SlimefunMocks.mockSlimefunItem(plugin, "CATEGORY_ITEMS_TEST_ITEM", new CustomItem(Material.BAMBOO, "&6Test Bamboo"));
+        item.setCategory(category);
+        item.register(plugin);
+        item.load();
+
+        Assertions.assertEquals(1, category.getItems().size());
+
+        // Size must still be 1 since we disallow duplicates
+        item.setCategory(category);
```

Comment:
You should also check size after this first one. You're checking the intended initial behaviour and then a checked behaviour (dupe)

Reply:
I know I could, doing it twice was just an exaggeration